### PR TITLE
140799931 kh trail day

### PR DESCRIPTION
**Pull request by kheppenstall**
-Trail day create with two part to choose location on a map
-Trail day show page

@ski-climb Going to add a background worker in the next PR.

### CODE REVIEW

File: app/views/admin/trail_days/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<%= content_for(:mapbox) do %>
+  <%= javascript_include_tag 'https://api.mapbox.com/mapbox-gl-js/v0.32.1/mapbox-gl.js' %>
+<% end %>
+
+<div id='map' class='map-trail-day-locator' data-env=<%= ENV["MAPBOX_ACCESS_TOKEN"] %>>
```
Comment (ski-climb) [inline app/views/admin/trail_days/edit.html.erb]: looking at this again - I don't think we're actually using the MAPBOX_ACCESS_TOKEN.  I think that was left over from when we were experimenting with a few different ways to get the token to Mapbox.  I bet we can take that out of all of our views - including the original view where I accidentally left it in.

File: app/views/shared/_nav.html.erb [hunk 2]
```diff
@@ -27,6 +27,11 @@
           <%= link_to 'Login with Strava', '/auth/strava' %>
         </li>
       <% end %>
+      <% if current_admin %>
```
Comment (ski-climb) [inline app/views/shared/_nav.html.erb]: Nice.  I agree - I think this is still small enough that we don't need to separate out into different navbars yet.

File: spec/factories/trail_days.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+FactoryGirl.define do
+  factory :trail_day do
+    participant_email_addresses { Faker::Internet.email + ', ' + Faker::Internet.email }
```
Comment (ski-climb) [inline spec/factories/trail_days.rb]: well thought out - nice.

File: spec/features/trail_day/create_spec.rb [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+      it 'redirects to edit page' do
+        fill_in 'trail_day[participant_email_addresses]', with: attributes[:participant_email_addresses]
+        fill_in 'trail_day[duration_in_hours]', with: attributes[:duration_in_hours]
+        fill_in 'trail_day[description]', with: attributes[:description]
+        click_on 'Move on to select the location'
+
+        trail_day = TrailDay.first
+
+        attributes.each { |name, value| expect(trail_day.send(name)).to eq value }
```
Comment (ski-climb) [inline spec/features/trail_day/create_spec.rb]: 👍

File: spec/features/trail_day/create_spec.rb [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        expect(current_path).to eq edit_admin_trail_day_path(trail_day)
+        expect(page).to have_content 'Please select the trail day starting location by dragging the blue dot clicking submit location.'
```
Comment (ski-climb) [inline spec/features/trail_day/create_spec.rb]: maybe "... dragging the blue dot AND clicking submit..."

**Comment by ski-climb** [comment]
I'd say we can merge this in as a functioning piece of code.  One small thing - let's format the start date/time for the trail day show page.  Also, what would you think about restricting the 'minutes' portion of the form field to only allow something like 0, 15, 30, 45 minutes?

File: app/views/admin/trail_days/edit.html.erb (hunk 1 above)
Comment (kheppenstall) [inline app/views/admin/trail_days/edit.html.erb]: Ok I'll remove that.

File: spec/features/trail_day/create_spec.rb (hunk 5 above)
Comment (kheppenstall) [inline spec/features/trail_day/create_spec.rb]: Nice call I can make that change.

**Comment by kheppenstall** [comment]
@ski-climb Right now the date time looks like this:
screen shot 2017-03-01 at 12 18 01 pm

Thoughts on how we might clean that up?

**Comment by kheppenstall** [comment]
@ski-climb I'm happy to work on restricting minutes a bit later on after we get some of the things we need for the assessment tomorrow down.

**Pull request merged**